Content.Server/Botany/Systems/MutationSystem.cs

```diff
@@ -0,0 +1,211 @@
+        MutateFloat(ref seed.Potency               , 30f   , 100f , 5 , totalbits , 2*severity);
+
+        // Kill the plant (30)
+        MutateBool(ref seed.Viable         , false , 30 , totalbits , severity);
+
+        // Fun (70)
+        MutateBool(ref seed.Seedless       , true  , 10 , totalbits , severity);
+        MutateBool(ref seed.Slip           , true  , 10 , totalbits , severity);
+        MutateBool(ref seed.Sentient       , true  , 10 , totalbits , severity);
+        MutateBool(ref seed.Ligneous       , true  , 10 , totalbits , severity);
+        MutateBool(ref seed.Mysterious     , true  , 10 , totalbits , severity);
+        MutateBool(ref seed.Bioluminescent , true  , 10 , totalbits , severity);
+        seed.BioluminescentColor = RandomColor(seed.BioluminescentColor, 10, totalbits, severity);
+    }
+
+    public static SeedData Cross(SeedData a, SeedData b)
+    {
+        SeedData result = b.Clone();
+
+        result.Chemicals = random(0.5f) ? a.Chemicals : result.Chemicals;
+
+        CrossFloat(ref result.NutrientConsumption, a.NutrientConsumption);
+        CrossFloat(ref result.WaterConsumption, a.WaterConsumption);
+        CrossFloat(ref result.IdealHeat, a.IdealHeat);
+        CrossFloat(ref result.HeatTolerance, a.HeatTolerance);
+        CrossFloat(ref result.IdealLight, a.IdealLight);
+        CrossFloat(ref result.LightTolerance, a.LightTolerance);
+        CrossFloat(ref result.ToxinsTolerance, a.ToxinsTolerance);
+        CrossFloat(ref result.LowPressureTolerance, a.LowPressureTolerance);
+        CrossFloat(ref result.HighPressureTolerance, a.HighPressureTolerance);
+        CrossFloat(ref result.PestTolerance, a.PestTolerance);
+        CrossFloat(ref result.WeedTolerance, a.WeedTolerance);
+
+        CrossFloat(ref result.Endurance, a.Endurance);
+        CrossInt(ref result.Yield, a.Yield);
+        CrossFloat(ref result.Lifespan, a.Lifespan);
+        CrossFloat(ref result.Maturation, a.Maturation);
+        CrossFloat(ref result.Production, a.Production);
+        CrossFloat(ref result.Potency, a.Potency);
+
+        CrossBool(ref result.Seedless, a.Seedless);
+        CrossBool(ref result.Viable, a.Viable);
+        CrossBool(ref result.Slip, a.Slip);
+        CrossBool(ref result.Sentient, a.Sentient);
+        CrossBool(ref result.Ligneous, a.Ligneous);
+        CrossBool(ref result.Bioluminescent, a.Bioluminescent);
+        result.BioluminescentColor = random(0.5f) ? a.BioluminescentColor : result.BioluminescentColor;
+
+        // Hybrids have a high chance of being seedless. Balances very
+        // effective hybrid crossings.
+        if (a.Name == result.Name && random(0.7f))
+        {
+            result.Seedless = true;
+        }
+
+        return result;
+    }
+
+    // Mutate reference 'val' between 'min' and 'max' by pretending the value
+    // is representable by a thermometer code with 'bits' number of bits and
+    // randomly flipping some of them.
```

Comment:
is it a thermometer code if you're 'flipping' them? i thought thermometer codes were unary?

Reply:
The underlying idea is that if you have a trait representable by '11100' (a true thermometer code), flipping one bit gives, as an example, '11101' (not a thermometer code), which you can sort and get '1110' (a thermometer code again).

Statistically, this gives an expected final distribution that trends to the middle of where the thermometer code is. And it is also a thermometer code in the sense that the minimum change is one least-significant bit. But it might just be poor naming.